### PR TITLE
Create scorecards.yml

### PR DESCRIPTION
**Pull request by mlieberman85**
Add OpenSSF scorecard github code checker.

Signed-off-by: Michael Lieberman <mlieberman85@gmail.com>

### CODE REVIEW

File: .github/workflows/scorecards.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8  # v3.1.0
+        with:
+          persist-credentials: false
+
+      - name: "Run analysis"
+        uses: ossf/scorecard-action@15c10fcf1cf912bd22260bfec67569a359ab87da  # v2.1.1
+        with:
+          results_file: results.sarif
+          results_format: sarif
+          # (Optional) "write" PAT token. Uncomment the `repo_token` line below if:
```
Comment (mlieberman85) [inline .github/workflows/scorecards.yml:44]: I don't know how we want to handle this as the token would belong to just one user. In the past I've handled this through service accounts and stuff.

**Comment by pxp928** [comment]
closes #370

**Review by bradbeck — APPROVED** [review]

**Review by pxp928 — APPROVED** [review]

File: .github/workflows/scorecards.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+          # - you want to enable the Branch-Protection check on a *public* repository, or
+          # - you are installing Scorecards on a *private* repository
+          # To create the PAT, follow the steps in https://github.com/ossf/scorecard-action#authentication-with-pat.
+          repo_token: ${{ secrets.SCORECARD_TOKEN }}
```
Comment (rgreinho) [inline .github/workflows/scorecards.yml:48]: Would the `${{ secrets.GITHUB_TOKEN }}` work here instead?

**Comment by bradbeck** [comment]
@mlieberman85 You need `Signed-off-by: ...` in each commit.

File: .github/workflows/scorecards.yml (hunk 2 above)
Comment (bradbeck) [reply .github/workflows/scorecards.yml:48]: Looks like that's what they do here: https://github.com/apache/commons-imaging/blob/master/.github/workflows/scorecards-analysis.yml#L54

Comment (rgreinho) [reply .github/workflows/scorecards.yml:48]: Nope, it is a token that @mlieberman85 generated, instead of the one created by the CI.

**Comment by mlieberman85** [comment]
Ughh, forgot that editing in Github doesn't automatically add the DCO.

**Review by bradbeck — APPROVED** [review]

**Review by pxp928 — APPROVED** [review]
LGTM

**Pull request merged**